nbconvert/filters/markdown.py

```diff
@@ -100,4 +102,21 @@ def markdown2rst(source, extra_args=None):
     out : string
       Output as returned by pandoc.
     """
-    return convert_pandoc(source, 'markdown', 'rst', extra_args=extra_args)
+    def rawlatex2math_hook(obj):
+        if obj.get('t') == 'RawBlock' and obj['c'][0] == 'latex':
+            obj['t'] = 'Para'
+            obj['c'] = [{
+                't': 'Math',
+                'c': [
+                    {'t': 'DisplayMath', 'c': []},
+                    obj['c'][1],
+                ]
+            }]
+        return obj
+
+    def rawlatex2math(text):
+        json_data = json.loads(text, object_hook=rawlatex2math_hook)
+        return json.dumps(json_data)
+    
+
+    return convert_pandoc(source, 'markdown', 'rst', extra_args=extra_args) 
```

Comment:
Did you mean to use the new `pandoc_with_filter` function you defined here? At the moment, I don't think it's using the functions functions you define above.